test/downstream-project-cmake-test/CMakeLists.txt

```diff
@@ -13,5 +13,35 @@ else()
   message(STATUS "The current exported EDM4HEP version is " ${EDM4HEP_VERSION})
 endif()
 
+# Use the EDM4HEP_DATA_DIR to derive extended data model
+message(STATUS "${EDM4HEP_DATA_DIR}")
+if ("${EDM4HEP_DATA_DIR}" STREQUAL "")
+  message(WARNING "Unable to extend data model since no installed yaml file found.")
+else()
+  message(STATUS "Current EDM4HEP data model is available at ${EDM4HEP_DATA_DIR}.")
+  PODIO_GENERATE_DATAMODEL(edm4dis edm4dis.yaml headers sources
+    UPSTREAM_EDM edm4hep:${EDM4HEP_DATA_DIR}/edm4hep.yaml
+    OUTPUT_FOLDER ${CMAKE_CURRENT_BINARY_DIR}
+  )
+  add_library(edm4dis SHARED
+    ${sources}
+  )
+
+  target_link_libraries(edm4dis
+    PUBLIC EDM4HEP::edm4hep
+    PUBLIC podio::podio
+  )
+  target_include_directories(edm4dis
+    PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+    $<INSTALL_INTERFACE:include>
+  )
+
+  PODIO_GENERATE_DICTIONARY(edm4dis ${headers} 
+    SELECTION ${CMAKE_CURRENT_BINARY_DIR}/src/selection.xml
+    OPTIONS --library ${CMAKE_SHARED_LIBRARY_PREFIX}edm4dis${CMAKE_SHARED_LIBRARY_SUFFIX}
+  )
+  set_target_properties(edm4dis-dictgen PROPERTIES EXCLUDE_FROM_ALL TRUE)
```

Comment:
In this case it probably won't make too much of a difference, since you have to link against edm4hep in any case, but just in case you are not aware of this, there is a `PODIO_ADD_DATAMODEL_CORE_LIB` and a `PODIO_ADD_ROOT_IO_DICT` macro that handles this, see, e.g. how they are used in edm4hep:
https://github.com/key4hep/EDM4hep/blob/fb3d92b3f6c47584f3d3256706d400bde5344ecc/edm4hep/CMakeLists.txt#L4-L8

Definitions/documentation:
- [`PODIO_ADD_DATAMODEL_CORE_LIB`](https://github.com/AIDASoft/podio/blob/e576e36c8bd9dd90ec27bf96166a4d5b891148e6/cmake/podioMacros.cmake#L204)
- [`PODIO_ADD_ROOT_IO_DICT`](https://github.com/AIDASoft/podio/blob/e576e36c8bd9dd90ec27bf96166a4d5b891148e6/cmake/podioMacros.cmake#L244)

Reply:
Hmm, tried that, but this now fails to include the generated header directory in the build directory on the compilation line.

Reply:
```console
[ 12%] Building CXX object CMakeFiles/edm4dis.dir/src/InclusiveKinematics.cc.o
/home/wdconinc/git/EDM4hep/test/downstream-project-cmake-test/build/src/InclusiveKinematics.cc:4:10: fatal error: edm4dis/InclusiveKinematics.h: No such file or directory
    4 | #include "edm4dis/InclusiveKinematics.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
Marked as draft since it's not working right now.